endoreg_db/serializers/video_segmentation.py

```
@@ -0,0 +1,479 @@
+                    ).first()
+
+                    if existing_segment:
+                        # If a segment with the same start_frame exists but the end_frame is different, update it
+                        if float(existing_segment.end_frame_number) != end_frame:
+                            existing_segment.end_frame_number = end_frame
+                            existing_segment.save()
+                            updated_segments.append(existing_segment)
+                    else:
+                        # If no existing segment matches, create a new one
+                        new_entries.append(
+                            LabelRawVideoSegment(
```

Ruff (0.8.2)

Undefined name `LabelRawVideoSegment`

(F821)